apps/web/src/modules/create-proposal/components/TransactionForm/Migration/Migration.tsx

```diff
@@ -0,0 +1,172 @@
+import { Stack } from '@zoralabs/zord'
+import axios from 'axios'
+import { FormikHelpers } from 'formik'
+import { useRouter } from 'next/router'
+import { ReactNode } from 'react'
+import useSWR from 'swr'
+import { encodeFunctionData, parseAbi } from 'viem'
+import { useBalance, useContractRead } from 'wagmi'
+
+import { L1_MESSENGERS, L2_DEPLOYMENT_ADDRESSES } from 'src/constants/addresses'
+import { auctionAbi, tokenAbi } from 'src/data/contract/abis'
+import { DaoMember } from 'src/data/subgraph/requests/memberSnapshot'
+import { TransactionType } from 'src/modules/create-proposal/constants/transactionType'
+import { useProposalStore } from 'src/modules/create-proposal/stores'
+import { prepareMerkle } from 'src/modules/create-proposal/utils/prepareMerkle'
+import { useDaoStore } from 'src/modules/dao/stores/useDaoStore'
+import { useChainStore } from 'src/stores/useChainStore'
+import { AddressType, CHAIN_ID } from 'src/typings'
+
+import { MigrationFormC1 } from './MigrationFormC1'
+import { MigrationFormC2 } from './MigrationFormC2'
+import { MigrationFormC2Values } from './MigrationFormC2.schema'
+import { MigrationTracker } from './MigrationTracker'
+import { prepareMigrationDeploy } from './prepareMigrationDeploy'
+
+type MembersQuery = {
+  membersList: DaoMember[]
+}
+
+export const Migration: React.FC = () => {
+  const chain = useChainStore((x) => x.chain)
+  const thisDao = useDaoStore()
+  const thisChain = useChainStore()
+  const { treasury, auction, token } = useDaoStore((x) => x.addresses)
+  const addTransaction = useProposalStore((state) => state.addTransaction)
+  const { isReady } = useRouter()
+
+  const { data: treasuryBalance } = useBalance({
+    address: treasury as `0x${string}`,
+    chainId: chain.id,
+  })
+
+  const { data: paused } = useContractRead({
+    abi: auctionAbi,
+    address: auction,
+    functionName: 'paused',
+    chainId: chain.id,
+  })
+
+  const { data: existingFounders } = useContractRead({
+    abi: tokenAbi,
+    address: token,
+    functionName: 'getFounders',
+    chainId: chain.id,
+  })
+
+  let merkleRoot: `0x${string}`
+
+  const {
+    data: members,
+    error,
+    isValidating,
+  } = useSWR(isReady ? [token, chain.id] : undefined, () =>
+    axios
+      .get<MembersQuery>(
+        `/api/memberSnapshot/${token}?chainId=${
+          chain.id
+        }&page=${undefined}&limit=${10000}`
+      )
+      .then(async (x) => {
+        x.data.membersList
+        merkleRoot = await prepareMerkle(x.data.membersList)
+      })
+  )
+
+  const deployed = false
+
+  let daoProgress = 0
+  if (paused) {
```

Comment:
try not to nest if statements like this. makes it a lot harder to read. instead do something like:

```js
const treasuryMigrated = treasuryBalance && treasuryBalance.value < 1000000000000000;
let daoProgress = 3

if(!paused) daoProgress = 0; // DAO needs to pause auctions
else if(!deployed) daoProgress = 1; // DAO needs to be deployed on L2
else if(!treasuryMigrated) daoProgress = 2; // Treasury needs to be migrated
```

also might be more readable to use an enum for DAO progress (gets hard to remember what 1, 2, 3 mean)

```js
enum DAOMigrationProgress {
  Default = 0
  Paused = 1
  Deployed = 2
  Finalized = 3
}
```

Reply:
ah yes the enum is a great idea, and good call on flipping the code to use just a series of ifs